slo-monitor/src/monitors/pod_monitor.go

```diff
@@ -133,6 +139,15 @@ func (pm *PodStartupLatencyDataMonitor) Run(stopCh chan struct{}) {
 			pm.handlePodUpdate(p)
 			return nil
 		},
+		func(obj interface{}) error {
+			p, ok := obj.(*v1.Pod)
+			if !ok {
+				glog.Errorf("Failed to cast observed object to *v1.Pod: %v", obj)
+				return nil
+			}
```

Comment:
I hoped that it's done by `DeletionHandlingMetaNamespaceKeyFunc`
 https://github.com/kubernetes/kubernetes/blob/28123f959c995ef13e21b16b966abcafae56c69d/staging/src/k8s.io/client-go/tools/cache/controller.go#L213